### PR TITLE
Chore: Make some improvements

### PR DESCRIPTION
**Pull request by sarvaje**
## Pull request checklist

Make sure you:

- [x] Signed the [Contributor License Agreement](https://cla.js.foundation/sonarwhal/sonarwhal)
- [x] Followed the [commit message guidelines](https://sonarwhal.com/docs/contributor-guide/getting-started/pull-requests/#commit-messages)

For non-trivial changes, please make sure you also:

- [x] Added/Updated related documentation.
- [x] Added/Updated related tests.

### CODE REVIEW

**Comment by molant** [comment]
@sarvaje can you rebase so I can have a better idea of what is being changed?

**Comment by sarvaje** [comment]
@molant @alrra this should be ready to review

File: packages/rule-html-checker/src/rule.ts [hunk 1]
```diff
@@ -67,9 +67,14 @@ export default class HtmlCheckerRule implements IRule {
         let ignoredMessages;
         /** The options to pass to the HTML checker. */
         const scanOptions = {
-            data: '',
-            format: 'json',
-            validator: ''
+            body: '',
+            headers: {
+                'Content-Type': 'text/html; charset=utf-8',
+                'User-Agent': 'Sonarwhal'
```
Comment (molant) [inline packages/rule-html-checker/src/rule.ts]: `sonarwhal`

File: packages/rule-html-checker/src/rule.ts [hunk 2]
```diff
@@ -129,28 +129,39 @@ export default class HtmlCheckerRule implements IRule {
...
-            const { response } = data;
-
+        const requestRetry = async (options, retries: number = 3) => {
             /*
              * HACK: Need to do a require here in order to be capable
              * of mocking when testing the rule and `import` doesn't
              * work here.
              */
-            const htmlChecker = require('html-validator');
+            const { requestAsync, delay } = require('sonarwhal/dist/src/lib/utils/misc');
```
Comment (molant) [inline packages/rule-html-checker/src/rule.ts]: With TypeScript 2.8 we should be capable of doing dynamic imports. Can you please test it and see if it works?

**Comment by sarvaje** [comment]
@molnat done

**Comment by sarvaje** [comment]
@molant I have added a new commit

**Comment by molant** [comment]
@sarvaje are you going to replace all `require` with dynamic imports or are you leaving that for another PR?

**Comment by sarvaje** [comment]
@molant @alrra there is another commit with some dynamic imports :)

**Comment by alrra** [comment]
@sarvaje Can you fix the merge conflicts? Thanks!

**Comment by sarvaje** [comment]
@alrra done!

**Pull request closed without merging**